src/index_intf.ml

```diff
@@ -170,6 +174,10 @@ module type S = sig
   val sync : t -> unit
   (** [sync t] syncs a read-only index with the files on disk. Raises
       {!RW_not_allowed} if called by a read-write index. *)
+
+  val is_merging : t -> bool
+  (** [is_merging t] returns true if [t] is running an async merge. Raises
```

Comment:
```suggestion
  (** [is_merging t] returns true if [t] is running a merge. Raises
```

I don't think it technically has to be an async merge: it could be blocking in another thread owned by the user.